diff -r -u ./algorithm ../result/algorithm
--- ./algorithm	Wed May 20 12:02:31 2009
+++ ../result/algorithm	Wed May 20 12:01:23 2009
@@ -880,6 +880,11 @@
                                RandomAccessIterator last, Pointer buffer,
                                Distance buffer_size, T*, Compare comp);
 
+#if !defined(__MINMAX_DEFINED) 
+  template <class T>
+  inline const T& (min)(const T& a, const T& b);
+#endif
+
   template <class RandomAccessIterator, class Pointer, class Distance, class T>
   inline void __stable_sort (RandomAccessIterator first,
                              RandomAccessIterator last,
Common subdirectories: ./iso and ../result/iso
diff -r -u ./istream ../result/istream
--- ./istream	Wed May 20 12:02:31 2009
+++ ../result/istream	Wed May 20 12:01:23 2009
@@ -67,7 +67,7 @@
 
     _EXPLICIT basic_istream(basic_streambuf<charT, traits> *sb);
 
-    _EXPLICIT basic_istream(EmptyCtor) : basic_ios<charT, traits>(1) {}
+    _EXPLICIT basic_istream(ios_base::EmptyCtor) : basic_ios<charT, traits>(1) {}
     virtual ~basic_istream();
 
     class sentry 
@@ -307,7 +307,7 @@
           }
           else
           {
-            ostream_type* ost = stream.tie();
+            basic_ostream<charT, traits>* ost = stream.tie();
             if (ost && ost->rdbuf() != sb) 
               ost->flush();
 
@@ -419,9 +419,11 @@
         const istream_iterator<T,charT,traits,Distance>& x,
         const istream_iterator<T,charT,traits,Distance>& y);
 #else
-    friend inline bool operator== <> (
+/* SMOSS 29/08/07 inline not allowed in friend declaration */
+    friend /*inline*/ bool operator== <> (
         const istream_iterator<T,charT,traits,Distance>& x,
         const istream_iterator<T,charT,traits,Distance>& y);
+/* END SMOSS 29/08/07 inline not allowed in friend declaration */
 #endif
   public: 
     typedef charT char_type;
diff -r -u ./math.h ../result/math.h
--- ./math.h	Wed May 20 12:02:31 2009
+++ ../result/math.h	Wed May 20 12:01:23 2009
@@ -313,6 +313,11 @@
 #endif
 /* END adopted by C99 */
 
+/* SMOSS 29/08/07 <floatingpoint.h> conflicts with stdlib_iso.h */
+#ifdef __EXTENSIONS__
+# undef __EXTENSIONS__
+#endif //__EXTENSIONS__
+/* END SMOSS 29/08/07 <floatingpoint.h> conflicts with stdlib_iso.h */
 #if defined(__EXTENSIONS__) || !defined(__cplusplus)
 #include <floatingpoint.h>
 #endif
@@ -319,6 +324,11 @@
 #endif	/* defined(__EXTENSIONS__) || !defined(_XOPEN_SOURCE) */
 #endif	/* defined(__EXTENSIONS__) || defined(_XOPEN_SOURCE) || ... */
 
+/* SMOSS 29/08/07 <floatingpoint.h> conflicts with stdlib_iso.h */
+#ifdef __EXTENSIONS__
+# define __EXTENSIONS__
+#endif //__EXTENSIONS__
+/* END SMOSS 29/08/07 <floatingpoint.h> conflicts with stdlib_iso.h */
 #if defined(__cplusplus) && defined(__GNUC__)
 #undef	exception
 #endif
diff -r -u ./ostream ../result/ostream
--- ./ostream	Wed May 20 12:02:31 2009
+++ ../result/ostream	Wed May 20 12:01:23 2009
@@ -88,7 +88,7 @@
     typedef basic_ios<charT, traits>               ios_type;
 
     _EXPLICIT basic_ostream(basic_streambuf<charT, traits> *sb);
-    _EXPLICIT basic_ostream(EmptyCtor) : basic_ios<charT, traits>(1) {}
+    _EXPLICIT basic_ostream(ios_base::EmptyCtor) : basic_ios<charT, traits>(1) {}
     virtual ~basic_ostream();
 
     class sentry {
@@ -97,8 +97,7 @@
       sentry(basic_ostream<charT,traits>& stream)
         : __stream(stream)                  
       {  
-                      
-        streambuf_type* sb = stream.rdbuf();
+        basic_streambuf<charT, traits>* sb = stream.rdbuf();
         if ( sb ) {
           stream.rdbuf()->buffer_mutex_.acquire();
         }
Common subdirectories: ./rw and ../result/rw
diff -r -u ./streambuf ../result/streambuf
--- ./streambuf	Wed May 20 12:02:31 2009
+++ ../result/streambuf	Wed May 20 12:01:23 2009
@@ -927,7 +927,10 @@
    * int_type snextc()
    * returns the next character
    */
-
+/* SMOSS 29/08/07 Can't use typename in template specializations */
+#ifdef _TYPENAME
+# define _TYPENAME
+#endif
   inline _TYPENAME basic_streambuf<char, char_traits<char> >::int_type
   basic_streambuf<char, char_traits<char> >::snextc()
   {
@@ -1208,6 +1211,11 @@
   {
     return seekpos(sp, which);
   }
+#ifdef _TYPENAME
+# define _TYPENAME typename
+# undef _OLD_TYPENAME
+#endif
+/* END SMOSS 29/08/07 Can't use typename in template specializations */
 
   /*
    * basic_streambuf *pubsetbuf(char_type *, streamsize)
diff -r -u ./string ../result/string
--- ./string	Wed May 20 12:02:31 2009
+++ ../result/string	Wed May 20 12:01:23 2009
@@ -797,7 +797,9 @@
     // Null string ref
     //
 #ifndef _RWSTD_NO_STATIC_DEF3
-    static __SUNW_GLOBAL const __null_ref_type __nullref;
+/* SMOSS 29/08/07 Can't resolve #define __SUNW_GLOBAL __global */
+    static /* __SUNW_GLOBAL */ const __null_ref_type __nullref;
+/* SMOSS 29/08/07 Can't resolve #define __SUNW_GLOBAL __global */
 #endif
 
     static __string_ref_type * __getNullRep ()
Common subdirectories: ./sys and ../result/sys
diff -r -u ./typeinfo ../result/typeinfo
--- ./typeinfo	Wed May 20 12:02:31 2009
+++ ../result/typeinfo	Wed May 20 12:01:23 2009
@@ -34,7 +34,8 @@
 	// Implementation
 	const void* __data;
 	type_info(const void*) throw();
-	friend __Cimpl::__type_info_friend;
+	/* SMOSS 29/08/07 Can't resolve */
+	//friend __Cimpl::__type_info_friend;
     };
 
     class bad_typeid: public exception {
diff -r -u ./iso/stdlib_iso.h ../result/iso/stdlib_iso.h
--- ./iso/stdlib_iso.h	Wed May 20 12:02:31 2009
+++ ../result/iso/stdlib_iso.h	Wed May 20 12:01:23 2009
@@ -110,8 +110,10 @@
 extern double atof(const char *);
 extern int atoi(const char *);
 extern long int atol(const char *);
+/* SMOSS 29/08/07
 extern void *bsearch(const void *, const void *, size_t, size_t,
 	int (*)(const void *, const void *));
+*/
 #if __cplusplus >= 199711L
 extern "C++" {
 	void *bsearch(const void *, const void *, size_t, size_t,
@@ -131,7 +133,6 @@
 extern size_t mbstowcs(wchar_t *_RESTRICT_KYWD, const char *_RESTRICT_KYWD,
 	size_t);
 extern int mbtowc(wchar_t *_RESTRICT_KYWD, const char *_RESTRICT_KYWD, size_t);
-extern void qsort(void *, size_t, size_t, int (*)(const void *, const void *));
 #if __cplusplus >= 199711L
 extern "C++" {
 	void qsort(void *, size_t, size_t, int (*)(const void *, const void *));
diff -r -u ./rw/iterator ../result/rw/iterator
--- ./rw/iterator	Wed May 20 12:02:31 2009
+++ ../result/rw/iterator	Wed May 20 12:01:23 2009
@@ -426,10 +426,12 @@
     friend inline difference_type (std::operator-) (const self& x, const self& y);
     friend inline self (std::operator+)     (difference_type n, const self& x);
 #else
-    friend inline bool operator== <>   (const self& x, const self& y);
-    friend inline bool operator<  <>   (const self& x, const self& y);
-    friend inline difference_type operator- <> (const self& x, const self& y);
-    friend inline self operator+  <>   (difference_type n, const self& x);
+/* SMOSS 29/08/07 removing 'inline' from friend declarations */
+    friend /*inline*/ bool operator== <>   (const self& x, const self& y);
+    friend /*inline*/ bool operator<  <>   (const self& x, const self& y);
+    friend /*inline*/ difference_type operator- <> (const self& x, const self& y);
+    friend /*inline*/ self operator+  <>   (difference_type n, const self& x);
+/* END SMOSS 29/08/07 removing 'inline' from friend declarations */
 #endif        
   protected:
 
@@ -581,7 +583,9 @@
     typedef Reference reference;
     typedef Pointer pointer;
 
-    friend inline bool operator== <>   (const self& x, const self& y);
+/* SMOSS 29/08/07 removing 'inline' from friend declarations */
+    friend /*inline*/ bool operator== <>   (const self& x, const self& y);
+/* END SMOSS 29/08/07 removing 'inline' from friend declarations */
         
   protected:
 
diff -r -u ./sys/regset.h ../result/sys/regset.h
--- ./sys/regset.h	Wed May 20 12:02:31 2009
+++ ../result/sys/regset.h	Wed May 20 12:13:34 2009
@@ -16,7 +16,10 @@
 #pragma ident	"@(#)regset.h	1.9	06/05/12 SMI"
 
 #include <sys/feature_tests.h>
-
+/* SMOSS 29/08/07     */
+//#include <inttypes.h>
+typedef long long unsigned int uint64_t;
+/* end SMOSS 29/08/08 */
 #if !defined(_ASM)
 #include <sys/types.h>
 #endif
