Index: assert.h
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/assert.h,v
retrieving revision 1.1
retrieving revision 1.2
diff -c -3 -p -r1.1 -r1.2
*** assert.h	1 Nov 2002 14:00:28 -0000	1.1
--- assert.h	31 Mar 2003 14:52:29 -0000	1.2
*************** _CRTIMP void __cdecl _assert(void *, voi
*** 53,58 ****
  }
  #endif
  
! #define assert(exp) (void)( (exp) || (_assert(#exp, __FILE__, __LINE__), 0) )
  
  #endif  /* NDEBUG */
--- 53,58 ----
  }
  #endif
  
! #define assert(exp) (void)( (exp) || (_assert((void*)#exp, __FILE__, __LINE__), 0) )
  
  #endif  /* NDEBUG */
Index: typeinfo.h
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/typeinfo.h,v
retrieving revision 1.1
diff -c -3 -p -r1.1 typeinfo.h
*** typeinfo.h	1 Nov 2002 14:00:28 -0000	1.1
--- typeinfo.h	17 May 2004 14:28:54 -0000
***************
*** 37,42 ****
--- 37,43 ----
  #endif  /* _DLL */
  #endif  /* _CRTIMP */
  
+ namespace std {
  class type_info {
  public:
      _CRTIMP virtual ~type_info();
*************** private:
*** 51,57 ****
      type_info(const type_info& rhs);
      type_info& operator=(const type_info& rhs);
  };
! 
  
  // This include must occur below the definition of class type_info
  #include <stdexcpt.h>
--- 52,59 ----
      type_info(const type_info& rhs);
      type_info& operator=(const type_info& rhs);
  };
! }
! using std::type_info;
  
  // This include must occur below the definition of class type_info
  #include <stdexcpt.h>
Index: xlocale
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/xlocale,v
retrieving revision 1.1
retrieving revision 1.2
diff -c -3 -p -r1.1 -r1.2
*** xlocale	1 Nov 2002 14:00:28 -0000	1.1
--- xlocale	1 Nov 2002 16:01:12 -0000	1.2
***************
*** 13,18 ****
--- 13,21 ----
  #pragma warning(push,3)
  
  _STD_BEGIN
+ class _CRTIMP2 locale;
+ template<class _Elem> class collate;
+ template<class _Facet> const _Facet& __cdecl use_facet(const locale& _Loc);
  
  		// TEMPLATE CLASS _Locbase
  template<class _Dummy>
Index: xstddef
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/xstddef,v
retrieving revision 1.1
retrieving revision 1.3
diff -c -3 -p -r1.1 -r1.3
*** xstddef	1 Nov 2002 14:00:28 -0000	1.1
--- xstddef	18 Mar 2003 20:42:54 -0000	1.3
*************** _STD_BEGIN
*** 22,28 ****
   #define _RERAISE	throw
  
   #define _THROW0()	throw ()
!  #define _THROW1(x)	throw (...)
   #define _THROW(x, y)	throw x(y)
   #else /* no exceptions */
   #define _TRY_BEGIN	{{
--- 22,28 ----
   #define _RERAISE	throw
  
   #define _THROW0()	throw ()
!  #define _THROW1(x)	throw (x)
   #define _THROW(x, y)	throw x(y)
   #else /* no exceptions */
   #define _TRY_BEGIN	{{
*************** _STD_BEGIN
*** 42,48 ****
   #define _TEMPLATE	template<>
  
   #ifndef _TEMPLATE_STAT
!   #define _TEMPLATE_STAT	template<>
   #endif
  
   #ifndef _TEMPLATE_MEMBER
--- 42,48 ----
   #define _TEMPLATE	template<>
  
   #ifndef _TEMPLATE_STAT
!   #define _TEMPLATE_STAT
   #endif
  
   #ifndef _TEMPLATE_MEMBER
Index: yvals.h
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/yvals.h,v
retrieving revision 1.1
retrieving revision 1.2
diff -c -3 -p -r1.1 -r1.2
*** yvals.h	1 Nov 2002 14:00:28 -0000	1.1
--- yvals.h	24 Jan 2004 18:03:00 -0000	1.2
*************** _STD_END
*** 93,100 ****
  		/* VC++ COMPILER PARAMETERS */
  #define _LONGLONG	__int64
  #define _ULONGLONG	unsigned __int64
! #define _LLONG_MAX	0x7fffffffffffffff
! #define _ULLONG_MAX	0xffffffffffffffff
  
  		/* INTEGER PROPERTIES */
  #define _C2			1	/* 0 if not 2's complement */
--- 93,100 ----
  		/* VC++ COMPILER PARAMETERS */
  #define _LONGLONG	__int64
  #define _ULONGLONG	unsigned __int64
! #define _LLONG_MAX	0x7fffffffffffffffLL
! #define _ULLONG_MAX	0xffffffffffffffffLL
  
  		/* INTEGER PROPERTIES */
  #define _C2			1	/* 0 if not 2's complement */
