Index: typeinfo
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/typeinfo,v
retrieving revision 1.1
retrieving revision 1.3
diff -c -3 -p -r1.1 -r1.3
*** typeinfo	1 Nov 2002 14:00:28 -0000	1.1
--- typeinfo	11 Feb 2003 18:58:12 -0000	1.3
***************
*** 31,38 ****
   #include <typeinfo.h>
  
   _STD_BEGIN
   #if _HAS_EXCEPTIONS
- using ::type_info;
  using ::bad_cast;
  using ::bad_typeid;
  using ::__non_rtti_object;
--- 31,51 ----
   #include <typeinfo.h>
  
   _STD_BEGIN
+ class type_info {
+ public:
+     _CRTIMP virtual ~type_info();
+     _CRTIMP int operator==(const type_info& rhs) const;
+     _CRTIMP int operator!=(const type_info& rhs) const;
+     _CRTIMP int before(const type_info& rhs) const;
+     _CRTIMP const char* name() const;
+     _CRTIMP const char* raw_name() const;
+ private:
+     void *_m_data;
+     char _m_d_name[1];
+     type_info(const type_info& rhs);
+     type_info& operator=(const type_info& rhs);
+ };
   #if _HAS_EXCEPTIONS
  using ::bad_cast;
  using ::bad_typeid;
  using ::__non_rtti_object;
Index: xlocale
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/xlocale,v
retrieving revision 1.1
retrieving revision 1.2
diff -c -3 -p -r1.1 -r1.2
*** xlocale	1 Nov 2002 14:00:28 -0000	1.1
--- xlocale	1 Nov 2002 16:01:12 -0000	1.2
***************
*** 13,18 ****
--- 13,21 ----
  #pragma warning(push,3)
  
  _STD_BEGIN
+ class _CRTIMP2 locale;
+ template<class _Elem> class collate;
+ template<class _Facet> const _Facet& __cdecl use_facet(const locale& _Loc);
  
  		// TEMPLATE CLASS _Locbase
  template<class _Dummy>
Index: xstddef
===================================================================
RCS file: /cvsroot/GxInclude/Vc7/Include/xstddef,v
retrieving revision 1.1
retrieving revision 1.2
diff -c -3 -p -r1.1 -r1.2
*** xstddef	1 Nov 2002 14:00:28 -0000	1.1
--- xstddef	1 Nov 2002 16:01:29 -0000	1.2
*************** _STD_BEGIN
*** 22,28 ****
   #define _RERAISE	throw
  
   #define _THROW0()	throw ()
!  #define _THROW1(x)	throw (...)
   #define _THROW(x, y)	throw x(y)
   #else /* no exceptions */
   #define _TRY_BEGIN	{{
--- 22,28 ----
   #define _RERAISE	throw
  
   #define _THROW0()	throw ()
!  #define _THROW1(x)	throw (x)
   #define _THROW(x, y)	throw x(y)
   #else /* no exceptions */
   #define _TRY_BEGIN	{{
